shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1808,24 +1818,19 @@ - (void)performOrientationUpdate:(UIInterfaceOrientationMask)new_preferences {
     _orientationPreferences = new_preferences;
 
     if (@available(iOS 16.0, *)) {
+#if APPLICATION_EXTENSION_API_ONLY
+      UIWindowScene* windowScene = self.viewIfLoaded.window.windowScene;
+      [self performOrientationUpdateOnWindowScene:windowScene];
+#else
       for (UIScene* scene in UIApplication.sharedApplication.connectedScenes) {
         if (![scene isKindOfClass:[UIWindowScene class]]) {
           continue;
         }
-        UIWindowScene* windowScene = (UIWindowScene*)scene;
-        UIWindowSceneGeometryPreferencesIOS* preference =
-            [[UIWindowSceneGeometryPreferencesIOS alloc]
-                initWithInterfaceOrientations:_orientationPreferences];
-        [windowScene
-            requestGeometryUpdateWithPreferences:preference
-                                    errorHandler:^(NSError* error) {
-                                      os_log_error(OS_LOG_DEFAULT,
-                                                   "Failed to change device orientation: %@",
-                                                   error);
-                                    }];
-        [self setNeedsUpdateOfSupportedInterfaceOrientations];
+        [self performOrientationUpdateOnWindowScene:(UIWindowScene*)scene];
       }
+#endif
     } else {
+#if !APPLICATION_EXTENSION_API_ONLY
       UIInterfaceOrientationMask currentInterfaceOrientation =
           1 << [[UIApplication sharedApplication] statusBarOrientation];
```

Comment:
If `available` you could check the `self.viewIfLoaded.window.windowScene.statusBarOrientation ` since this was deprecated in iOS 13.
```
@property(readonly, nonatomic) UIInterfaceOrientation statusBarOrientation API_UNAVAILABLE(tvos) API_DEPRECATED("Use the interfaceOrientation property of the window scene instead.", ios(2.0, 13.0));
```